packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js

```diff
@@ -85,14 +87,28 @@ export const ConditionBuilderItem = ({
     }
   }, [contentRef, open]);
 
+  const eligibleToClose = () => {
+    if (
+      event?.target?.closest?.(`.${carbonPrefix}--date-picker__calendar`) ||
+      Array.from(event?.target?.classList)?.includes(
+        `.${carbonPrefix}--date-picker__calendar`
+      )
+    ) {
+      return false;
+    }
+    return true;
+  };
+
   return (
     <Popover
       open={open}
       isTabTip
       role="gridcell"
       className={popOverClassName}
       onRequestClose={() => {
-        setOpen(false);
+        if (eligibleToClose()) {
```

Comment:
lets not add logic like this in the HTML. i would create an onClose handler that deals with all this.

Reply:
resolved